### DISCUSSION TITLE
Bricked Intel I219 LM

### DISCUSSION THREAD

**Discussion by kuro68k** [ipxe/ipxe#582] [Q&A]
Sorry this isn't directly ipxe related, but searching for solutions has lead me here and it's possible that the same tools used to load ipxe can be used to fix my Intel I219 LM.

It appears that the flash memory for this integrated NIC has become corrupted somehow. The VID is 8086 (Intel), but the PID is FFFF. As such drivers don't attach to it. I tried Intel's bootutil too, but it doesn't detect the NIC. FreeBSD and Windows both identify it as a network device, but no more.

Is there some way to fix it? Some tool I can use to flash the firmware? Where do I even get the firmware?

Any hints greatly appreciated.

**Comment by NiKiZe** [comment]
You need to differentiate between firmware/ROM which for an onboard NIC is located in the bios/system firmware, and any possible configuration flash, which might be embedded in NIC chipset itself.

So I find it unlikely that the Intel ROM flash tools are usable, (integrated NIC don't have flashrom as mentioned above)

Could you tell us more about the device? Do you have board information?
And how did you end up in current state?

I have seen something similar before, at those time it has been machines hit by lightning or other powersurges thru the Ethernet port.

**Comment by kuro68k** [comment]
Thanks, that's enlightening. The machine is a Dell OptiPlex 7040 with Intel Q170 chipset.

I'm not entirely sure how it happened. I bought an Intel i225 2.5G ethernet card in M.2 form factor for that machine, but it wasn't working. I saw that there were firmware updates available so tried one, and it failed to properly erase the flash memory and bricked the board. I fixed it by de-soldering the flash memory and programming it with the firmware image in a stand alone programmer, and then re-attaching it. It works fine now.

However, after doing that the on-board I219 NIC stopped working. Previously it was detected and a driver attached by FreeBSD. So I'm wondering if the Intel firmware update tool managed to erase the wrong flash memory, but that seems unlikely given what you say about it not having any.

I tried updating the machine with Dell's latest BIOS, which is actually from January 12th this year, but it didn't help.
